assets/js/blocks/collection-filters/utils.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { ProductCollectionQuery } from '@woocommerce/blocks/product-collection/types';
+
+function taxonomyMapper( taxonomy: string ) {
+	const map: Record< string, string > = {
```

Comment:
I don't think you need to define this type, right? TS will infer it better if you don't.

Reply:
Ah yea, that's because `taxonomy` is a `string` you can either do:

```ts
function taxonomyMapper( taxonomy: 'product_tag' | 'product_cat' )
```

or

```ts
return map[ taxonomy as keyof typeof map ];
```

I think both of those should work. The first one is more elegant, IMHO.

Reply:
> function taxonomyMapper( taxonomy: 'product_tag' | 'product_cat' )

Besides `product_tag` and `product_cat`, `taxonomy` can be a custom taxonomy. Doing this way makes TS happy and the code works as expected, but it makes TS and developers think that the function only accepts `product_tag` or `product_cat` as the valid parameter, which is not true. I feel like we're tricking the compiler here.

> return map[ taxonomy as keyof typeof map ];

Do you mean `return map[ taxonomy as keyof typeof map ] || _unstable_tax_${ taxonomy };`? If yes, then I don't think typecasting is more elegant than using the current tbh, when custom taxonomy is passed, it's not a key of the `map``.

Reply:
> If yes, then I don't think typecasting is more elegant than using the current tbh, when custom taxonomy is passed, it's not a key of the map.

It is, in a way. Because when it's not, we are in the right side of the operator. It's a bit of a compromise. If you wanted to be fully clean, you can use a type guard or a conditional type. That'd be the 100% clean solution.